src/Billing/Models/PayPalIPNTransactionModel.cs

```diff
@@ -72,6 +73,12 @@ public PayPalIPNTransactionModel(string formData)
             OrganizationId = organizationId;
         }
 
+        if (metadata.TryGetValue("provider_id", out var providerIdStr) &&
+            Guid.TryParse(organizationIdStr, out var providerId))
```

Comment:
I think this is supposed to be `providerIdStr`?

Reply:
Good catch! Pushed